sample.lua

```diff
@@ -13,12 +13,23 @@ cmd:option('-temperature', 1)
 cmd:option('-gpu', 0)
 cmd:option('-gpu_backend', 'cuda')
 cmd:option('-verbose', 0)
+cmd:option('-seed', 0)
 local opt = cmd:parse(arg)
 
 
 local checkpoint = torch.load(opt.checkpoint)
 local model = checkpoint.model
 
+if opt.seed == 0 then
+  opt.seed = torch.random()
+end
+torch.manualSeed(opt.seed)
```

Comment:
Might be simpler to force a manual seed only if `-seed` argument is set:

```
if <flag set>:
  torch.manualSeed(...)
end
```

Reply:
I agree that this code looks odd.  The reason I did this is to make sure that if you see some output that you would like to see again, and if you had the verbose flag set to report the seed, you can re-run by then including the seed - even if you didn't originally set the seed flag.  So the code always sets a seed - using your provided value if you set one and a random number if you didn't (or specified zero!).  I thought this was clever ;-)

Reply:
Ah okay that makes sense.